gluon-mesh-vpn-wireguard/files/lib/gluon/gluon-mesh-wireguard/checkuplink

```diff
@@ -52,63 +59,36 @@ if [ "$(uci get wireguard.wireguard.enabled)" == "true" ] || [ "$(uci get wiregu
-                	uci show wireguard.peer_$ANZAHL 2>/dev/null > /dev/null                                    
-                	if [ $? != 0 ]; then                                                                       
-                        	break                                                                              
-                	fi                                                                                         
-        	done                                                                                               
-        	ANZAHL=`expr $ANZAHL - 1`                                                                          
-                                                                                                           
-        	#Und welchen nehmen wir jetzt??                                                                    
-        	SPEED1=0                                                                                           
-        	I=0                                                                                                
-        	until [ $I -eq $ANZAHL ]                                                                           
-        	do                                                                                                 
-                	I=`expr $I + 1`                                                                            
-                	#Von allen Servern die durschnittliche potenzielle Geschwindigkeit pro Wireguard-Verbindung
-                	URL=$(uci get wireguard.peer_$I.endpoint | cut -d':' -f1)                        
-			SPEED2=$(curl https://$URL/speed.php -k)                                           
-                	if [ $SPEED2 -gt $SPEED1 ]              # Wenn aktueller Server hoehren Wert hat,
-                        		then AUSWAHL=$I                 # Servernummer speichern                 
-                        	SPEED1=$SPEED2                                                           
-                	fi                                                                               
-        	done                                                                                     
-        	URL=$(uci get wireguard.peer_$AUSWAHL.endpoint | cut -d':' -f1)                          
+		modprobe wireguard                                                       
+
+                NUMBER_OF_PEERS=$(uci show wireguard | egrep -e peer_[0-9].endpoint | wc -l)
+                PEER=$(awk 'BEGIN{srand();print int(rand()*($NUMBER_OF_PEERS)+1) }')
                                                                                                  
 		#Die alten peers loeschen
 		wg set $INTERFACE peer  $(wg |  awk 'BEGIN {RS=""} /endpoint/ {print $2}') remove
-                batctl if del gre
-                ip link delete dev gre
+                batctl if del vxlan-wan
+                ip link delete dev vxlan-wan
```

Comment:
I guess `vx_mesh_vpn` would be closer to Gluons naming?

Reply:
Will rename the whole stuff to this :)